debugger/src/debug_adapter/repl_commands.rs

```diff
@@ -0,0 +1,774 @@
+use super::{
+    dap_adapter::{disassemble_target_memory, DapStatus},
+    dap_types::{
+        CompletionItem, CompletionItemType, CompletionsArguments, DisassembledInstruction,
+        EvaluateArguments, EvaluateResponseBody,
+    },
+};
+use crate::{
+    debugger::{
+        core_data::CoreHandle, debug_entry::DebugSessionStatus, session_data::BreakpointType,
+    },
+    DebuggerError,
+};
+use probe_rs::{debug::VariableName, CoreStatus, HaltReason, MemoryInterface};
+use std::{fmt::Display, time::Duration};
+
+pub(crate) enum ReplCommandArgs {
+    Required(&'static str),
+    Optional(&'static str),
+}
+
+impl Display for ReplCommandArgs {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ReplCommandArgs::Required(arg_value) => {
+                write!(f, "{arg_value}")
+            }
+            ReplCommandArgs::Optional(arg_value) => {
+                write!(f, "[{arg_value}]")
+            }
+        }
+    }
+}
+
+/// The handler is a function that takes a reference to the target core, and a reference to the response body.
+/// The response body is used to populate the response to the client.
+/// The handler can return a [`DebugSessionStatus`], which is used to determine if the debug session should continue, or if it should be terminated.
+/// The handler can also return a [`DebuggerError`], which is used to populate the response to the client.
+/// The majority of the REPL command results will be populated into the response body.
+type H = fn(
```

Comment:
Please rename this, I find it very confusing that this is called the same as the generic parameter to the struct in line 47.

Reply:
Good catch ... I meant to clean that up.